### PR TITLE
Add a new endpoint for listing collects

### PR DESCRIPTION
**Pull request by yulintan**

### CODE REVIEW

File: collect.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"fmt"
+	"time"
+)
+
+const collectsBasePath = "admin/collects"
+const collectsResourceName = "collects"
+
+// CollectService is an interface for interfacing with the collect endpoints
+// of the Shopify API.
+// See: https://help.shopify.com/api/reference/product/collect
```
Comment (umsu2) [inline collect.go]: that url is invalid, should be https://help.shopify.com/en/api/reference/products/collect

File: collect_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+func TestCollectCount(t *testing.T) {
+	setup()
+	defer teardown()
+
+	httpmock.RegisterResponder("GET", "https://fooshop.myshopify.com/admin/collects/count.json",
+		httpmock.NewStringResponder(200, `{"count": 5}`))
+
+	httpmock.RegisterResponder("GET", "https://fooshop.myshopify.com/admin/collects/count.json?since_id=123",
```
Comment (umsu2) [inline collect_test.go]: should try to use RegisterResponderWithQuery?
https://github.com/jarcoal/httpmock/blob/v1/transport.go

File: collect.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package goshopify
+
+import (
+	"fmt"
+	"time"
+)
+
+const collectsBasePath = "admin/collects"
+const collectsResourceName = "collects"
```
Comment (umsu2) [inline collect.go]: this is not used anywhere?

File: collect.go (hunk 1 above)
Comment (yulintan) [reply collect.go]: good catch!

**Pull request merged**